homeassistant/components/axis/config_flow.py

```diff
@@ -72,6 +75,7 @@ async def async_step_user(self, user_input=None):
                 )
 
                 self.serial = device.vapix.serial_number
+                assert self.serial
```

Comment:
I'm in doubt this should be asserted. I think it's just initialized incorrectly.

Reply:
`format_mac` doesn't like `None`, so we need to ensure it `is not None`.
I've adjusted the code slightly to use an intermediate variable